lib/caasp.pm

```diff
@@ -158,10 +158,27 @@ sub check_reboot_changes {
 sub trup_install {
     # rebootmgr has to be turned off as prerequisity for this to work
     script_run "rebootmgrctl set-strategy off";
-    my $package = shift;
-    trup_call("pkg install $package");
-    process_reboot 1;
-    assert_script_run("rpm -qi $package");
+    my ($input, $unnecessary, $necessary, $pkg_is_present) = shift;
+    my ($pkg_required, $not_needed) = 0;
+    my @pkg = split(' ', $input);
+    foreach (@pkg) {
+        $pkg_is_present = script_run("rpm -q $_") == 0;
+        if ($pkg_is_present) {
```

Comment:
Why not to encapsulate condition here as it's used only once? 
```perl
if(script_run("rpm -q $_") == 0) 
```
or `unless(script_run("rpm -q $_"))` or `if(!script_run("rpm -q $_"))` or `if(script_run("! rpm -q $_"))`

Reply:
Originally I had it like that. Then I decided that the `if` statement looks more readable that way:

```
if ($pkg_is_present) {
```
instead of:
```
if(script_run("rpm -q $_") == 0) 
```

Sure, I can encapsulate it but in that case I would like to put a comment on top:

```perl
        # if package is present
        if (!script_run("rpm -q $_")) {
```

Do you prefer that way?

Reply:
In general it's matter of taste. But here it helps to have less variables which may confuse code reader. Cheers.